[Group Work] Config in .JSON (#3229)
PrairieLearn/PrairieLearn: docs/assessment.md

## Enabling group work for collaborative assessments

By default, assessment instances are tied to only one user. By setting `groupWork: true`, multiple students will be able to work on the same assessment instance.

> Instructor should also set information about group size constraints and student permissions in the json format. Example:

Reviewer: ```suggestion
Information about the group configuration can be set in the `infoAssessment.json` file. For example:
```

```json
{
        "groupWork": true,
        "groupMaxSize": 6,
        "groupMinSize": 2,
        "studentGroupCreate": true,
        "studentGroupJoin": true,
        "studentGroupLeave": true,
}

> ```

Reviewer: Can we add a table listing all the options here, their types, default values, and descriptions? This would be similar to the tables of options here: https://prairielearn.readthedocs.io/en/latest/elements/#customizations

## Enabling group work for collaborative assessments

By default, assessment instances are tied to only one user. By setting `groupWork: true`, multiple students will be able to work on the same assessment instance.
Information about the group configuration can be set in the `infoAssessment.json` file. For example:
```json
{
        "groupWork": true,
        "groupMaxSize": 6,
        "groupMinSize": 2,
        "studentGroupCreate": true,
        "studentGroupJoin": true,
        "studentGroupLeave": true,
}
```
Attribute | Type | Default | Description
--- | --- | --- | ---
`groupWork` | boolean | false | Enable the group work for the assessment.

> `groupMaxSize` | integer | - | The maximum size of a group.
> `groupMinSize` | integer | - | The minimum size of a group.

Reviewer: ```suggestion
`groupMaxSize` | integer | - | The maximum size of a group (default: no minimum).
`groupMinSize` | integer | - | The minimum size of a group (default: no maximum).
```
